### PR TITLE
Prevents duplicate notifications

### LINKED ISSUE #3896: The notification messages in Bridge duplicate when the version check toggle is updated
Your issue may already be reported! Please search on the [issue tracker](https://github.com/keptn/keptn/issues) before creating a new one.

**Environment**

* Client OS (e.g., Linux, macOS, Windows):  MacOS
* Keptn Version (`keptn version`): 0.81
* Kubernetes Cloud Provider (e.g., GKE, AKS): k3s
* Kubernetes version (`kubectl version`):
-bash-4.2$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.10+k3s1", GitCommit:"72e8196c9b9eb277f56951db6d2d9e5f9f70feac", GitTreeState:"clean", BuildDate:"2021-04-16T18:36:38Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.21) and server (1.19) exceeds the supported minor version skew of +/-1

Chrome Version 90.0.4430.85 (Official Build) (x86_64)
**Affected Component**

* [ ] Docs
* [ ] CLI
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] Helm
* [ ] Istio
* [ X] Bridge
* [ ] Approval
* [ ] Datastore
* [ ] REST API
* [ ] eventbroker / distributor
* [ ] jmeter

**Describe the bug**

When you turn on and off the version check, the messages at the top get duplicated.

**To Reproduce**

1. Go to the profile page and turn off the version check
image

2. while on the profile page, turn version check back on
image
3. the message duplicates
4. keep repeating and more messages show up.
image
5. when you refresh the whole page the messages go back to just one set.

**Expected behavior**

* [ ] No duplicate messages
* [ ] When disabling the feature, the messages should disappear

**Current behavior**

**Screenshots**

**Additional context**

These are messages related to upgrading 0.81 to 0.82 -- so it is unknown if this is related to just these messages  or to all messages.

### PR DESCRIPTION
**Pull request by laneli**
Fixes #3896 
Signed-off-by: Elisabeth Lang <elisabeth.lang@dynatrace.com>

### CODE REVIEW

File: bridge/client/app/_services/notifications.service.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ export class NotificationsService {
 
   addNotification(type, message) {
     let notification = Notification.fromJSON({ type, message });
-    this._notifications.next([...this._notifications.getValue(), notification]);
+    const duplicateNotifications = this._notifications.getValue().filter(n => n.type === notification.type && n.message === notification.message);
+    if(duplicateNotifications.length === 0) {
+      this._notifications.next([...this._notifications.getValue(), notification]);
+    }
```
Comment (christian-kreuzberger-dtx) [inline bridge/client/app/_services/notifications.service.ts:24]: If I read this code in 1 month, I wouldn't understand why we are doing this here...

Please add some comments here (check if notification already exists), and reference the issue itself for future reference

Comment (jsoref) [reply bridge/client/app/_services/notifications.service.ts:24]: Also, you probably should add a ` ` -between `if` and `(` 😉

Comment (laneli) [reply bridge/client/app/_services/notifications.service.ts]: Added comments for clarification

File: bridge/client/app/_services/notifications.service.ts [hunk 2]
```diff
@@ -17,8 +17,15 @@ export class NotificationsService {
   }
 
   addNotification(type, message) {
-    let notification = Notification.fromJSON({ type, message });
-    this._notifications.next([...this._notifications.getValue(), notification]);
+    let notification = Notification.fromJSON({type, message});
+
+    // Check if the notification to add already exists (
```
Comment (christian-kreuzberger-dtx) [inline bridge/client/app/_services/notifications.service.ts:22]: There is an opening `(` at the end, please remove it

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**